File: flap/src/main/java/me/yifeiyuan/flap/internal/IComponentModel.java
package me.yifeiyuan.flap.internal;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
/**
 * Flap Github: <a>https://github.com/AlanCheen/Flap</a>
 * todo
 * @author 程序亦非猿 [Follow me](<a> https://github.com/AlanCheen</a>)
 * @since 2020/3/24 4:35 PM
 * @since 1.5
 */
public interface IComponentModel {

    boolean areItemsTheSame(@NonNull IComponentModel other);

    boolean areContentsTheSame(@NonNull IComponentModel other);

    @Nullable
    Object getChangePayload(@NonNull IComponentModel newItem);
}

File: flap/src/main/java/me/yifeiyuan/flap/IComponentModel.kt
package me.yifeiyuan.flap

/**
 * todo
 * Created by 程序亦非猿 on 2021/9/22.
 *
 * Flap Github: <a>https://github.com/AlanCheen/Flap</a>
 * @author 程序亦非猿 [Follow me](<a> https://github.com/AlanCheen</a>)
 * @since 2020/9/22
 * @since 3.0
 */
interface IComponentModel {
    fun areItemsTheSame(other: IComponentModel): Boolean
    fun areContentsTheSame(other: IComponentModel): Boolean
    fun getChangePayload(newItem: IComponentModel): Any?
}